src/vcpkg/commands.build.cpp

```diff
@@ -597,7 +597,7 @@ namespace vcpkg
         }
 
         const auto arch = to_vcvarsall_toolchain(pre_build_info.target_architecture, toolset, pre_build_info.triplet);
-        const auto target = to_vcvarsall_target(pre_build_info.cmake_system_name);
+        const auto target = to_vcvarsall_target(pre_build_info.cmake_system_name, pre_build_info.cmake_system_version);
```

Comment:
If you want this to select the Windows SDK, I believe you need something like this instead?

```c++
auto vcvarsall_options = toolset.vcvarsall_options;
if (!pre_build_info.cmake_system_version.empty()) {
    vcvarsall_options.push_back(fmt::format("-winsdk={}", pre_build_info.cmake_system_version));
}
```

based on

![vcvarsall help text](https://github.com/user-attachments/assets/3f2f7992-6f02-4844-9e12-ec0d4ed74cfd)

Reply:
I think "vcvarsall.bat" is enough. And I have test it locally.
Using Visual Studio 2022:
![image](https://github.com/user-attachments/assets/b7061a54-74fc-418f-bd28-77a3bbd7aea6)
Using Visual Studio 2019:
![image](https://github.com/user-attachments/assets/d82348d3-5f14-49cf-842c-f30a96e39e62)
I believe on the old or new version of visual studio, "vcvarsall.bat" can achieve the same function. 
Your suggestion about using "Toolset::vcvarsall_options" is more reasonable.